### ISSUE TITLE
Add way to log "Type of Evidence"

### ISSUE THREAD

**Issue by adricaswell** [LandSciTech/ccviR#155]
Suggestion by Ilona: In addition to the comment box in Vulnerability and Spatial Vulnerability Questions, add a way to log the type of evidence used to support the response. Give users type of evidence options to select from (we will still need to decide what types are specified - but something along the lines of "Peer Reviewed Literature" and "Expert Knowledge"). This will promt/ensure that users record this information. Provide users with a selection menu in the form of a list or a dropdown (Sarah will have a better idea of what is possible). Find balance between adding functionality to the app without making the UI cluttered.

**Comment by steffilazerte** [comment]
We have a draft of this working now (loading, saving, and summarizing)

Current options are "Literature", "Expert Opinion", "Spatial Analysis", "Spatial Analysis - ccviR" (automatically selected for spatial results) and "Other".

There is also a figure summarizing the evidence used at the end of the Results.

<img src = "https://github.com/user-attachments/assets/441e46cc-eadc-43d2-8b4b-20d90cb3b5b2" width = 70%>
<img src = "https://github.com/user-attachments/assets/2e860d40-8c90-4d8a-9a96-30048a7118b0" width = 70%>
<img src = "https://github.com/user-attachments/assets/20980e52-4c8f-4339-b873-0ae33a67dcb7" width = 50%>
<img src = "https://github.com/user-attachments/assets/bc8362c0-607f-4c50-beec-59d8d12c66ae" width = 70%>

I didn't include a tooltip for several reasons a) they're a bit messy to implement in bootstrap 3, b) I could do a button like the current info buttons, but I think that would start to look a bit messy, c) since every tooltip would say the same thing, it might be better to have a new "About" or "Help" page to quickly explain how the app works (independent of the vignettes). And this would be a good spot for the evidence descriptions

### Questions

- Are these required inputs?
- Is it okay that I used selectors rather than checkboxes (as requested)?
- In the report, the figure could be a plotly figure if there's a need to be able to see which questions had which answers?
- How do you feel about tooltips vs. buttons vs. aout/Help page?

**Comment by see24** [comment]
Yes selectors are fine but it will need to be possible to select multiple, which is a bit easier to see with checkboxes. So it might be worth changing to checkboxes? what do you think?
Ya I think they should be required, or maybe have expert opnion selected as a default and then the user can change it if they use others. (Partially because I don't want to have to check every single box to run through it!)

A summary of evidence types is good in the report, but the table in the downloaded report should also list the evidence type. 
The meaning of the evidence types can just be defined somewhere no need for tool tips

**Comment by steffilazerte** [comment]
Okay, I've made some changes to this and then consequently to the index tab

**Multiple types of evidence**  
- I switched to a multi input selector to save on space but allow multiple selections. I think it's not too clunky, but doesn't take up as much space as a bunch of checkboxes. 
- The figure also includes a note about the average number of 'types of evidence' per question (this may not be necessary but I wanted some way of reconciling the difference between the number of questions answered and the counts. 

**Required evidence**  
I didn't end up making it required, but added a lot of notices where it was insufficient:   

- I added a running tally of evidence provided next to the running tally of questions answered. It only counts questions which have been answered (so adding evidence to an unanswered question has no effect). 
- I added evidence provided to the data completeness table
- I adjusted the evidence figure to clarify how many questions are still missing evidence 

**Printed report**

> A summary of evidence types is good in the report, but the table in the downloaded report should also list the evidence type.

You mentioned that the evidence should be included in the downloaded report. It is included in the list of answers at the end. Do you mean that the data completeness table should be included in the report? Or did you have another table in mind?

Finally, I tweaked the order of the results to show users the Data Completeness, regardless of whether they calculated the index. I'm not sure about this change, but since these sections aren't actually related to the index, but could be related to checking before calculating, I thought it might be worth exploring.

### At the start of the app, Data completeness is visible but incomplete
![Image](https://github.com/user-attachments/assets/c4046cae-afe0-41f5-bce1-17bfab575ebb)
 
### Once a previous run has been loaded, Data completeness is visible and complete, above the index results

![Image](https://github.com/user-attachments/assets/168b7298-f178-4406-895e-50542fcf145e)

**Comment by see24** [comment]
Looks good, just want to check that it works with the report once the issue with that is fixed #218

**Comment by see24** [comment]
The report worked and the type of evidence part looks correct as does the saved output. All done!!
